mne/tests/test_epochs.py

```diff
@@ -2450,6 +2450,48 @@ def test_events_list():
     assert_array_equal(epochs.events, np.array(events))
 
 
+def test_save_overwrite(tmpdir):
+    """Test saving with overwrite functionality."""
+
+    tempdir = str(tmpdir)
+    raw = mne.io.RawArray(np.random.RandomState(0).randn(100, 10000),
+                          mne.create_info(100, 1000.))
+
+    events = mne.make_fixed_length_events(raw, 1)
+    epochs = mne.Epochs(raw, events)
+
+    # scenario 1: overwrite=False (default) and there isn't a file to overwrite
+    # make a filename that has not already been saved to
+    fname1 = op.join(tempdir, 'test' + str(np.random.rand()) + '-epo.fif')
+    while op.isfile(fname1):
+        fname1 = op.join(tempdir, 'test' + str(np.random.rand()) + '-epo.fif')
+    # run function to be sure it doesn't throw an error
+    epochs.save(fname1)
+    # check that the file got written
+    assert(op.isfile(fname1))
+
+    # scenario 2: overwrite=False (default) and there is a file to overwrite
+    # fname1 exists because of scenario 1 above
+    pytest.raises(OSError, epochs.save, fname1)
```

Comment:
since MNE has now moved to python 3, why not use [FileExistsError](https://docs.python.org/3/library/exceptions.html#FileExistsError) ?

Reply:
I'm actually using `_check_fname` from utils (`check.py`), which is the same as what is used in `raw.save`. `_check_fname` raises an IOError. I changed my test accordingly now. Should I change `_check_fname` to use FileExistsError instead?

Reply:
FileExistsError is a subclass of OSError so this should be a backward-compatible change, so I'd say yes let's have `_check_fname` use `FileExistsError`. (You shouldn't need to update any existing `pytest.raises(OSError)` code because of the backward compatibility.)

Reply:
I would not change a function that has some history and is not broken/problematic.